lib/team.rb

```diff
@@ -0,0 +1,28 @@
+require 'csv'
+
+class Team
+
+  @@all = nil
+
+  def self.all
+    @@all
+  end
+
+  def self.from_csv(file_path)
+    csv = CSV.read("#{file_path}", headers: true, header_converters: :symbol)
+    @@all = csv.map do |row|
+      Team.new(row)
+    end
+  end
```

Comment:
Can be added into inheritance. functionality passes tests and creates objects which are stored into @@all which is an array

Reply:
+1, prevents the need to run the from_csv everytime we want to access the collection